cxx-qt-build/src/lib.rs

```diff
@@ -268,43 +238,74 @@ impl CxxQtBuilder {
         self
     }
 
-    /// Choose to disable Qt support
-    ///
-    /// This will disable including cxx-qt-lib headers.
-    pub fn disable_qt(mut self) -> Self {
-        self.qt_enabled = false;
-        self
-    }
-
     /// Specify rust file paths to parse through the cxx-qt marco
     ///
     /// Currently the path should be relative to CARGO_MANIFEST_DIR
     pub fn file(mut self, rust_source: &'static str) -> Self {
         self.rust_sources.push(rust_source);
+        println!("cargo:rerun-if-changed={}", rust_source);
         self
     }
 
-    // TODO: support globs with files("src/**/*.rs")
-
     /// Perform the build task, for example parsing and generating sources
-    pub fn build(self) {
+    pub fn build(mut self) {
         // Set clang-format format
         if generate_format(self.cpp_format).is_err() {
             panic!("Failed to set clang-format.");
         }
 
-        // TODO: somewhere check that we don't have duplicate class names
-        // TODO: later use the module::object to turn into module/object.h
+        self.cc_builder.cpp(true);
+        // MSVC
+        self.cc_builder.flag_if_supported("/std:c++17");
+        self.cc_builder.flag_if_supported("/Zc:__cplusplus");
+        self.cc_builder.flag_if_supported("/permissive-");
+        // GCC + Clang
+        self.cc_builder.flag_if_supported("-std=c++17");
+
+        // TODO: let user specify additional Qt modules to link
```

Comment:
Should we do this now in this branch or the next one ?

Seems it'd just be like `qt_modules(mut self, &[&str])` then store those as a vec in the CxxQtBuilder, then use that list in this build method. It would default to [core,gui] and if the downstream developer writes their own it can replace them? (so this then allows them to clear the list too for non-Qt builds?)

Reply:
yes, this just wasn't a priority before getting CI working on every OS